File: PSET2/crack.c
#define _XOPEN_SOURCE
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <unistd.h>

char *crypt(const char *key, const char *salt);

int main(int argc, char * argv[]){
    if(argc != 2){
        printf("Please enter valid password\n");
        return 1;
    }
    else{
        string hashed = argv[1];
        char letters[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
        char test[4];
        for(int i = 0, l = strlen(letters); i < l; i++){
            test[0] = letters[i];
            for(int j = 0; i < l; j++){
                test[1] = letters[j];
                for(int k = 0; k < l; k++){
                    test[2] = letters[k];
                    for(int h = 0; h < l; h++){
                        test[3] = letters[h];
                        if(strcmp(hashed,crypt(test, "50")) == 0){
                          printf("Password: %s\n", test);
                          return 0;
                        }
                    }
                }
            }
        }
    }
}

File: PSET6/crack.py
# import
import crypt
import sys

# prompt user for hashed password
if len(sys.argv) != 2:
    print("Please enter valid password")
    exit(code=1)

# store hashed password
hashed = sys.argv[1]

# create string with every letters of the alphabet lowercase and uppercase
letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"

# iterate over letters in 4 loops to get all combination possible
for i in letters:
    str = i

    for j in letters:
        str = i + j

        for k in letters:
            str = i + j + k

            for l in letters:
                str = i + j + k + l

                # print password if found
                if hashed == crypt.crypt(str, "50"):
                    print(f"Password: {str}")
                    exit(code=0)
